src/modules/update/common/update-common.c

```diff
@@ -0,0 +1,488 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#define SOL_LOG_DOMAIN &_log_domain
+#include "sol-log-internal.h"
+SOL_LOG_INTERNAL_DECLARE_STATIC(_log_domain, "auto-update");
+
+#include <dirent.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "update-common.h"
+
+#include "sol-json.h"
+#include "sol-http-client.h"
+#include "sol-lib-loader.h"
+#include "sol-mainloop.h"
+#include "sol-message-digest.h"
+#include "sol-update-modules.h"
+#include "sol-util.h"
+
+void
+delete_handle(struct sol_update_handle *handle)
+{
+    free(handle->hash_algorithm);
+    free(handle->hash);
+    free(handle->url);
+    free(handle->file_path);
+    if (handle->timeout)
+        sol_timeout_del(handle->timeout);
+    if (handle->conn)
+        sol_http_client_connection_cancel(handle->conn);
+    free(handle);
+}
+
+static void
+free_response_members(struct sol_update_info *info)
+{
+    free((void *)info->version);
+    free((void *)info->url);
+    free((void *)info->hash);
+    free((void *)info->hash_algorithm);
+}
+
+static void
+task_check_response(void *data, const struct sol_http_client_connection *conn,
```

Comment:
yes, just the name is misleading. My suggestion is to make it a "mix-in", so you implement the http in its own file, the "file" (save/apply) to its own file, etc. Then one platform that will use "http + file" (ie: yours), you include both and use their symbols. Others could define "coap" or "oic", but would still use "file". Or go with nvram/flash/eeprom/whatever.